src/Event/EventInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Event;
+
+/**
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+interface EventInterface
+{
+    public function getData();
+
+    public function setData($data): void;
```

Comment:
If we introduce our own event system, I would like to use immutability (as in our metadata system).